### PR TITLE
API-599: add end-to-end tests for download and upload AssetReferenceFile and AssetVariationFile

### PR DESCRIPTION
**Pull request by Doodoune**

### CODE REVIEW

File: tests/Api/AssetReferenceFile/DownloadAssetReferenceFileIntegration.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+        $this->server->setResponseOfPath(
+            '/'. sprintf(AssetReferenceFileApi::ASSET_REFERENCE_FILE_DOWNLOAD_URI, 'ziggy', 'en_US'),
+            new ResponseStack(
+                new Response(file_get_contents($expectedFilePath), [], 201)
+            )
+        );
+
+        $api = $this->createClient()->getAssetReferenceFileApi();
+        $api->uploadForLocalizableAsset($expectedFilePath, 'ziggy', 'en_US');
```
Comment (ahocquard) [inline tests/Api/AssetReferenceFile/DownloadAssetReferenceFileIntegration.php]: why do you upload? It is in theory useless

File: tests/Api/AssetReferenceFile/DownloadAssetReferenceFileIntegration.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        );
+
+        $api = $this->createClient()->getAssetReferenceFileApi();
+        $api->downloadFromLocalizableAsset('ziggy', 'en_US');
+    }
+
+    /**
+     * @expectedException \Akeneo\Pim\ApiClient\Exception\NotFoundHttpException
+     */
+    public function test_download_From_not_localizable_asset_not_found()
```
Comment (ahocquard) [inline tests/Api/AssetReferenceFile/DownloadAssetReferenceFileIntegration.php]: from (caps lock)

File: tests/Api/AssetVariationFile/DownloadAssetVariationFileApiIntegration.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+        $this->server->setResponseOfPath(
+            '/'. sprintf(AssetVariationFileApi::ASSET_VARIATION_FILE_DOWNLOAD_URI, 'ziggy_certif', 'ecommerce', AssetVariationFileApi::NOT_LOCALIZABLE_ASSET_LOCALE_CODE),
+            new ResponseStack(
+                new Response(file_get_contents($expectedFilePath), [], 201)
+            )
+        );
+
+        $api = $this->createClient()->getAssetVariationFileApi();
+        $api->uploadForNotLocalizableAsset($expectedFilePath, 'ziggy_certif', 'ecommerce');
```
Comment (ahocquard) [inline tests/Api/AssetVariationFile/DownloadAssetVariationFileApiIntegration.php]: same here

**Review by ahocquard — APPROVED** [review]

File: tests/Api/AssetReferenceFile/DownloadAssetReferenceFileIntegration.php (hunk 1 above)
Comment (Doodoune) [reply tests/Api/AssetReferenceFile/DownloadAssetReferenceFileIntegration.php]: Yep... I remove it

File: tests/Api/AssetReferenceFile/DownloadAssetReferenceFileIntegration.php (hunk 2 above)
Comment (Doodoune) [reply tests/Api/AssetReferenceFile/DownloadAssetReferenceFileIntegration.php]: Done

File: tests/Api/AssetVariationFile/DownloadAssetVariationFileApiIntegration.php (hunk 3 above)
Comment (Doodoune) [reply tests/Api/AssetVariationFile/DownloadAssetVariationFileApiIntegration.php]: Same correction :)

**Review by LaurentPetard — APPROVED** [review]

**Pull request merged**